lib/ide/src/controller/graph.rs

```diff
@@ -0,0 +1,177 @@
+//! Graph Controller.
+//!
+//! This controller provides access to a specific graph. It lives under a module controller, as
+//! each graph belongs to some module.
+
+pub mod mock;
+
+use crate::prelude::*;
+
+use flo_stream::Subscriber;
+
+pub use double_representation::graph::Id;
+pub use controller::node::Position;
+pub use controller::notification;
+
+
+
+// ==============
+// === Errors ===
+// ==============
+
+#[derive(Clone,Debug,Fail)]
+#[fail(display = "Node by ID {} was not found.", _0)]
```

Comment:
I believe that "node by ID" is not correct grammar form. To be confirmed. But I bet it should be "Node with ID" instead.